diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6ca4fc..dbf4687 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,11 @@ if (ENABLE_MIR)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DMIR_SUPPORT")
 endif()
 
+option(BUILD_32BIT "Build for 32bit and apply some quirks" OFF)
+if (BUILD_32BIT)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBUILD_32BIT")
+endif()
+
 if (NOT BINDERFS_PATH)
   set(BINDERFS_PATH "/dev/binderfs")
 endif()
diff --git a/src/anbox/input/device.cpp b/src/anbox/input/device.cpp
index 699a52c..da1f628 100644
--- a/src/anbox/input/device.cpp
+++ b/src/anbox/input/device.cpp
@@ -58,8 +58,13 @@ void Device::send_events(const std::vector<Event> &events) {
     // NOTE: A bit dirty but as we're running currently a 64 bit container
     // struct input_event has a different size. We rebuild the struct here
     // to reach the correct size.
+#if defined(BUILD_32BIT)
+	std::uint32_t sec;
+	std::uint32_t usec;
+#else
     std::uint64_t sec;
     std::uint64_t usec;
+#endif
     std::uint16_t type;
     std::uint16_t code;
     std::uint32_t value;
